### ISSUE TITLE
ext_authz: Support emit dynamic metadata in CheckResponse/DeniedHttpResponse

### ISSUE THREAD

**Issue by erikbos** [envoyproxy/envoy#12671]
Envoy's recently added support for for adding dynamic metadata in `CheckResponse`'s `OkHttpResponse` is great! I would love to see it being able to add dynamic metadata in `DeniedHttpResponse` as well. (I overlooked this while https://github.com/envoyproxy/envoy/pull/11820 was WIP :( )

An example use case would be where the authorization service received correct credentials, that fact should be logged, while the  request gets denied. (e.g. customer provided credentials are correct but customer's status is deactivated)

**Comment by dio** [comment]
Sorry for this. Seems like we need to deprecate the added `OkHttpResponse` dynamic metadata in favor of `CheckResponse`'s.

**Comment by erikbos** [comment]
No worries, thanks for working on this!

**Comment by rockspore** [comment]
In my use case, the access log service is not seeing the `envoy.filters.http.ext_authz` key in the metadata when ext_authz server sends a deny response.

To make sure the problem is not from my implementation, may I confirm if ext_authz is actually emitting dynamic metadata on denial in your case, @erikbos ?

**Comment by erikbos** [comment]
Indeed, you're right: looks like `CheckResponse.DynamicMetadata` for a deny response does not yet emit the metadata. It does for an allowed request.

(tested on go-control-plane `v0.9.9-0.20201210154907-fd9021fe5dad` and `eb99bfc13150fe57405138af3a6f1751fad7d8e2/1.18.0-dev`)

**Comment by ravenblackx** [comment]
This was fixed in #15058, associated with #15031.
